Dockerfile

```diff
@@ -4,7 +4,8 @@ FROM nikolaik/python-nodejs:python3.11-nodejs20
 
 WORKDIR /home/pn/js-ui
 COPY ui/package.json ui/package-lock.json ui/npm-install.sh ui/pdf.worker.js.patch ./
-RUN /bin/bash npm-install.sh
+RUN --mount=type=cache,target=/root/.npm/_cacache,sharing=locked \
+    /bin/bash npm-install.sh
```

Comment:
Plain vanilla 'docker build' barfs on this.  I had hoped it would be ignored as in COPY --chmod, but alas.  I'm not sure what to do about this.  I think since this is open-source, we should make an effort to make it as easy as possible to work on with basic tooling.  How big is the speedup?  Can we have a separate 'Dockerfile.fast' or generate something via the C preprocessor and a Makefile?

Reply:
Pinged Jon on this.

Reply:
Didn't hear back from Jon. Rather than stall or argue, I move the Dockerfile to Dockerfile.buildx and wrote a script to derive the Dockerfile.build from it.